apps/examples/testcase/le_tc/network/itc_net_send.c

```diff
@@ -0,0 +1,185 @@
+/****************************************************************************
+*
+* Copyright 2017 Samsung Electronics All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+****************************************************************************/
+
+// @file itc_net_send.c
+// @brief ITC Test Case Example for send() API
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <pthread.h>
+
+#include "tc_internal.h"
+
+#define PORTNUM 1110
+#define MAXRCVLEN 20
+
+static pthread_mutex_t g_lock = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t g_cond;
+
+/**
+* @testcase            :itc_net_send_n_fd
+* @brief               :used to transmit a message to another socket
+* @scenario            :sending invalid file descriptor
+* @apicovered          :send()
+* @precondition        :None
+* @postcondition       :None
+*/
+static void itc_net_send_n_fd(void)
+{
+	char *msg = "Hello World !\n";
+	int ret = send(-1, msg, strlen(msg), 0);
+
+	TC_ASSERT_EQ("send", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase            :itc_net_send_n_msg
+* @brief               :used to transmit a message to another socket
+* @scenario            :sending NULL msg
+* @apicovered          :send()
+* @precondition        :None
+* @postcondition       :None
+*/
+static void itc_net_send_n_msg(int socketFD)
+{
+	char *msg = NULL;
+	int ret = send(socketFD, msg, strlen(msg), 0);
+
+	TC_ASSERT_EQ("send", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :server
+* @brief                :For client server scenario creating thread function
+* @scenario             :None
+* API's covered         :socket,bind,listen,close
+* Preconditions         :None
+* Postconditions        :None
+*/
+static void *server(void *args)
+{
+	pthread_mutex_lock(&g_lock);
```

Comment:
why did you call pthread_mutex_lock() at entry point of function. you can call it right before pthread_cond_signal()

Reply:
Fixed review comments